### ISSUE TITLE
Double loss computation

### ISSUE THREAD

**Issue by matthiasreisser** [toologicbv/meta_learner#3]
Not entirely sure about this but it seems to me as if you are computing the nll twice for every set of optimizee parameters:
First, you compute 
`loss = reg_funcs.compute_neg_ll(average_over_funcs=False, size_average=False)`
around line 266 in train_optimizer.py
Then you compute the parameter delta with the meta optimizer and based on the updated optimizee parameters you compute the new nll:
`loss_step = meta_optimizer.step_loss(reg_funcs, par_new, average_batch=True)`
around line 292 in train_optimizer.py.

Then, in the next iteration you compute the loss
`loss = reg_funcs.compute_neg_ll(average_over_funcs=False, size_average=False)` again, which is, as far as I understand, the same as what happened in line 292 in the previous iteration. 

It seems to me you could gain efficiency here if I understand correctly.

**Comment by toologicbv** [comment]
yes indeed this can be more efficient Matthias. But honestly, I haven't checked my code on efficiency and probably this is something I should do. Will adjust this.
